4_visualize/src/plot_results.R

```diff
@@ -0,0 +1,412 @@
+    labs(title= paste(sprintf("%s", unique(run_group$site_id)), sprintf('GCM: %s', unique(run_group$gcm)), sep='\n'),  y="Temperature (\u00b0C)") +
+    theme_bw() +
+    theme(
+      panel.border = element_blank(),
+      strip.text = element_text(size=14),
+      axis.text = element_text(size=12),
+      plot.title = element_text(size=16, face="bold", hjust=0.5),
+      axis.title.x = element_text(size=16),
+      axis.title.y = element_text(size=16),
+      legend.text = element_text(size=14),
+      legend.title = element_blank(),
+      legend.position = "top"
+    )
+  
+  # Save plot
+  outfile <- sprintf(outfile_template, unique(run_group$site_id), unique(run_group$gcm))
+  ggsave(filename=outfile, plot=doy_plot, dpi=300, width=10, height=6)
+  return(outfile)
+}
+
+#' @title Plot predicted surface, middle, and bottom temperatures in a 
+#' given lake, for all 6 GCMs
+#' @description For a single lake, all 6 GCMs and all 3 time periods, 
+#' plot surface, middle, and bottom temperature predictions in each year, 
+#' by DOY, as well as the average GLM prediction for each DOY in each 
+#' 20-year time period. Also plot whether or not the model predicted ice 
+#' on a given DOY for any GCM or for all 6 GCMs. The output plot is 
+#' faceted by period and by depth class
+#' @param lake_group a single group from the `p2_glm_uncalibrated_lake_groups`
+#' grouped version of the `p2_glm_uncalibrated_runs` output tibble subset 
+#' to the site_id, gcm, time_period, raw_meteo_fl, export_fl, and 
+#' export_fl_hash columns and grouped by site_id. Then filtered to 
+#' only groups for which glm_success==TRUE for all runs in that group. 
+#' The function maps over these groups.
+#' @param outfile_template The template filepath for the exported png, which
+#' is customized with the site_id
+#' @return The filepath of the exported png 
+plot_20yr_average_preds_ice <- function(lake_group, outfile_template) {
+  # Get glm preds for the current lake (wide format)
+  glm_preds <- get_site_preds(lake_group$raw_meteo_fl, lake_group$export_fl, lake_group$gcm)
+
+  # Munge them to long format
+  glm_preds_long <- munge_long(glm_preds)
+  
+  # Pull out ice predictions, and create offset temperature for plotting
+  glm_preds_ice <- glm_preds %>%
+    select(gcm, year, doy, period, date, ice, temp_0) %>%
+    rename(temperature = temp_0) %>% # could use actual surface temp
+    mutate(temperature = -1) %>% # artificially set to -1 so doesn't block surface temp pred
+    mutate(depth = 0) %>%
+    filter(!(year == 2000))
+  
+  # Get dates when all GCMs predicted ice
+  glm_preds_ice_mean <- glm_preds_ice %>%
+    group_by(depth, doy, period) %>%
+    summarize(ice_all_gcms = ifelse(sum(ice)==n(), TRUE, FALSE), mean_temp = -2) %>%
+    filter(ice_all_gcms ==TRUE)
+
+  # Filter ice predictions to only those dates when ice was present
+  glm_preds_ice <- glm_preds_ice %>%
+    filter(ice==TRUE)
+  
+  # Determine surface, middle, and bottom depths for plotting
```

Comment:
This repeats a lot to, so maybe there is a data munge step that should happen first?

Reply:
Then can be used by multiple plots.

Reply:
This is now only getting called twice -- once within `plot_20yr_preds_ice()` and once within `get_surface_middle_bottom_preds`. It makes sense to keep that call within `get_surface_middle_bottom_preds`. To pull it out of `plot_20yr_presds_ice` completely would mean making 4 new targets in `4_visualize.R` (a new target for glm preds and mean glm preds for both the lake-gcm specific and lake-specific plots), which doesn't seem worth it to me.